spec/integration/inventory_spec.rb

```diff
@@ -148,6 +148,82 @@ def fact_plan(name = 'facts')
     end
   end
 
+  context 'when adding targets to a group in a plan' do
+    let(:conn) { conn_info('ssh') }
+    let(:inventory) do
+      {
+        groups: [
+          {
+            name: 'foo',
+            nodes: [
+              {
+                name: 'foo_1'
+              }
+            ],
+            config: {
+              transport: 'local'
+            },
+            facts: {
+              parent: 'keep'
+            },
+            vars: {
+              parent: 'keep'
+            },
+            groups: [
+              {
+                name: 'add_me',
+                nodes: [
+                  {
+                    name: conn[:host]
+                  }
+                ],
+                config: {
+                  transport: conn[:protocol],
+                  conn[:protocol] => {
+                    user: conn[:user],
+                    port: conn[:port]
+                  }
+                },
+                facts: {
+                  added_group: 'keep'
+                },
+                vars: {
+                  added_group: 'keep'
+                }
+              }
+            ]
+          },
+          {
+            name: 'bar',
+            nodes: [
+              {
+                name: 'bar_1'
+              }
+            ],
+            config: {
+              transport: 'local'
+            },
+            facts: {
+              exclude: 'dont_inherit'
+            },
+            vars: {
+              exclude: 'dont_inherit'
+            }
+          }
+        ]
+      }
+    end
+
+    it 'sets a variable on the target' do
```

Comment:
I intend to clean this test setup up. I wanted to get something in to help with review.